xarray/core/variable.py

```diff
@@ -893,16 +893,27 @@ def reduce(self, func, dim=None, axis=None, keep_attrs=False,
         if dim is not None and axis is not None:
             raise ValueError("cannot supply both 'axis' and 'dim' arguments")
 
+        if 'cum' in func.__name__:
```

Comment:
@shoyer, I double-checked.  This appears to work for both dataarray and dataset but it would be good to have some tests to ensure this functionality works in the future.